core/src/test/java/org/infinispan/stream/BaseStreamTest.java

```diff
@@ -1413,6 +1598,59 @@ public void testDoubleForEachOrdered() {
       }
    }
 
+   private static class ForEachDoubleInjected<E> implements DoubleConsumer,
+           CacheAware<Double, String>, Serializable {
+      private transient Cache<?, ?> cache;
+      private final int cacheOffset;
+      private final int atomicOffset;
+
+      private ForEachDoubleInjected(int cacheOffset, int atomicOffset) {
+         this.cacheOffset = cacheOffset;
+         this.atomicOffset = atomicOffset;
+      }
+
+      @Override
+      public void injectCache(Cache<Double, String> cache) {
+         this.cache = cache;
+      }
+
+      @Override
+      public void accept(double value) {
+         Cache<?, ?> cache = getForEachObject(cacheOffset);
+         if (cache != null && this.cache != null && cache.getName().equals(this.cache.getName())) {
+            DoubleSummaryStatistics stats = (DoubleSummaryStatistics) getForEachObject(atomicOffset);
+            synchronized (stats) {
+               stats.accept(value);
+            }
+         }
+      }
+   }
+
+   public void testDoubleForEachCacheInjected() {
+      Cache<Double, String> cache = getCache(0);
+      int range = 10;
+      // First populate the cache with a bunch of values
+      DoubleStream.iterate(0.0, d -> d + .5).limit(10).boxed().forEach(i -> cache.put(i, i + "-value"));
+
+      assertEquals(range, cache.size());
+      CacheSet<Map.Entry<Double, String>> entrySet = cache.entrySet();
+
+      int cacheOffset = populateNextForEachStructure(cache);
+      int offset = populateNextForEachStructure(new DoubleSummaryStatistics());
+      try {
+         createStream(entrySet).mapToDouble(toDouble).forEach(new ForEachDoubleInjected(cacheOffset, offset));
+         DoubleSummaryStatistics stats = getForEachObject(offset);
+         assertEquals(2.25, stats.getAverage());
+         assertEquals(0.0, stats.getMin());
+         assertEquals(4.5, stats.getMax());
+         assertEquals(10, stats.getCount());
+         assertEquals(22.5, stats.getSum());
```

Comment:
Couldn't you have a single method that can work either with a `ForEachDoubleInjected` or with a regular `DoubleConsumer`? You could then even use [TestNG parameters](http://testng.org/doc/documentation-main.html#parameters-dataproviders) to avoid creating multiple test methods.